starter-code/app.js

```diff
@@ -1,50 +1,126 @@
-const express      = require('express');
-const path         = require('path');
-const favicon      = require('serve-favicon');
-const logger       = require('morgan');
+const express = require('express');
+const path = require('path');
+const favicon = require('serve-favicon');
+const logger = require('morgan');
 const cookieParser = require('cookie-parser');
-const bodyParser   = require('body-parser');
-const mongoose     = require("mongoose");
+const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+const expressLayouts = require('express-ejs-layouts');
+const session = require('express-session');
+const FbStrategy = require('passport-facebook').Strategy;
+const passport = require('passport');
+// const MongoStore = require('connect-mongo')(session);
+
+// const auth = require('./routes/auth');
 
 const app = express();
 
 // Controllers
-const siteController = require("./routes/siteController");
+const siteController = require('./routes/siteController');
 
 // Mongoose configuration
-mongoose.connect("mongodb://localhost/ibi-ironhack");
+mongoose.Promise = Promise;
+mongoose.connect('mongodb://localhost/ibi-ironhack', {
+  keepAlive: true,
+  reconnectTries: Number.MAX_VALUE
+});
 
 // view engine setup
 app.set('views', path.join(__dirname, 'views'));
 app.set('view engine', 'ejs');
 
 // uncomment after placing your favicon in /public
-//app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
+app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
 app.use(logger('combined'));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
 
+// -- session
+
+// app.use(session({
+//   store: new MongoStore({
+//     mongooseConnection: mongoose.connection,
+//     ttl: 24 * 60 * 60 // 1 day
+//   }),
+//   secret: 'foobar',
+//   resave: true,
+//   saveUninitialized: true,
+//   cookie: {
+//     maxAge: 24 * 60 * 60 * 1000
+//   }
+// }));
+
+// app.use(function (req, res, next) {
+//   app.locals.user = req.session.currentUser;
+//   next();
+// });
+
+// Passport
+
+passport.use(new FbStrategy({
```

Comment:
Better put the passport configuration in its own file (inside a folder like helpers, for example).